packages/contracts-bedrock/src/L1/LeaderElectionBatchInbox.sol

```diff
@@ -5,6 +5,38 @@ pragma solidity ^0.8.0;
 /// @notice Interface for implementing a leader election scheme
 
 abstract contract LeaderElectionBatchInbox {
+    enum LeaderStatusFlags {
+        Scheduled,
+        Unscheduled,
+        Invalid
+    }
+
+    struct Meta {
+        /// Frame metadata
+        bytes16 channelId;
+        uint16 frameNumber;
+        uint32 frameDataLength;
+        bool isLast;
+        /// The number of L2 blocks in this channel (including the current frame).
+        uint16 numL2Blocks;
+    }
+
+    /// @notice Allows to submit a batch. This function checks that the caller is the leader for the current block.
+    ///
+    ///         Checking that the metadata matches the frames must be done in the derivation pipeline. If there is
+    ///         a mismatch between the metadata and the frames, the frames must be discarded.
+    ///
+    ///         Implementations of this contract interface can enforce behaviour based on the metadata submitted
+    ///         to this function.
+    ///
+    /// @param _metas metadata for the frames.
+    /// @param _frames frames to be submitted.
+    function submit(Meta[] memory _metas, bytes calldata _frames) external {
+        // TODO: pass metadata to isCurrentLeader and nextBlockAsLeader
```

Comment:
I don't think `submit` will ever call `nextBlockAsLeader`. That's just for the batchers to use. It does need to get passed to isCurrentLeader.